File: core/src/nl/t64/game/rpg/screens/world/mapobjects/GameMapSpawnPoint.java
package nl.t64.game.rpg.screens.world.mapobjects;

import com.badlogic.gdx.maps.objects.RectangleMapObject;
import lombok.Getter;
import nl.t64.game.rpg.screens.world.entity.Direction;


public class GameMapSpawnPoint extends GameMapObject {

    private final String fromMapName;
    private final String fromMapLocation;
    @Getter
    private final Direction direction;

    public GameMapSpawnPoint(RectangleMapObject rectObject) {
        super.rectangle = rectObject.getRectangle();
        this.fromMapName = rectObject.getName();
        this.fromMapLocation = createFromMapLocation(rectObject);
        this.direction = createDirection(rectObject);
    }

    public boolean isInConnectionWith(GameMapRelocator portal) {
        return (fromMapName.equals(portal.getFromMapName()) &&
                fromMapLocation.equalsIgnoreCase(portal.getToMapLocation()));
    }

    public float getX() {
        return rectangle.getX();
    }

    public float getY() {
        return rectangle.getY();
    }

    private String createFromMapLocation(RectangleMapObject rectObject) {
        return rectObject.getProperties().get("type", "", String.class);
    }

    private Direction createDirection(RectangleMapObject rectObject) {
        return Direction.valueOf(rectObject.getProperties()
                                           .get("direction", "NONE", String.class)
                                           .toUpperCase());
    }

}

File: core/src/nl/t64/game/rpg/screens/world/mapobjects/GameMapSpawnPoint.kt
package nl.t64.game.rpg.screens.world.mapobjects

import com.badlogic.gdx.maps.objects.RectangleMapObject
import ktx.tiled.property
import ktx.tiled.type
import nl.t64.game.rpg.screens.world.entity.Direction


class GameMapSpawnPoint(rectObject: RectangleMapObject) : GameMapObject(rectObject.rectangle) {

    private val fromMapName: String = rectObject.name
    private val fromMapLocation: String = rectObject.type ?: ""
    val direction: Direction = Direction.valueOf(rectObject.property("direction", "NONE").uppercase())

    val x: Float get() = rectangle.getX()
    val y: Float get() = rectangle.getY()

    fun isInConnectionWith(portal: GameMapRelocator): Boolean {
        return fromMapName == portal.fromMapName &&
                fromMapLocation.equals(portal.toMapLocation, true)
    }

}
